test/auto-languageclient.test.ts

```diff
@@ -1,24 +1,132 @@
 import AutoLanguageClient from "../lib/auto-languageclient"
+import { projectPathToWorkspaceFolder, ServerManager } from "../lib/server-manager"
+import { FakeAutoLanguageClient } from "./helpers"
+import { dirname } from "path"
+
+function mockEditor(uri: string, scopeName: string): any {
+  return {
+    getPath: () => uri,
+    getGrammar: () => {
+      return { scopeName }
+    },
+  }
+}
 
 describe("AutoLanguageClient", () => {
+  describe("ServerManager", () => {
+    describe("WorkspaceFolders", () => {
+      let client: FakeAutoLanguageClient
+      let serverManager: ServerManager
+
+      function beforeEachCallback() {
+        atom.workspace.getTextEditors().forEach((editor) => editor.destroy())
+        atom.project.getPaths().forEach((project) => atom.project.removePath(project))
+        client = new FakeAutoLanguageClient()
+        client.activate()
+
+        /* eslint-disable-next-line dot-notation */
+        serverManager = client["_serverManager"]
+      }
+
+      async function afterEachCallback() {
+        serverManager.stopListening()
+        await serverManager.stopAllServers()
```

Comment:
It was working with Sinon. I think this is related to the way jasmine mocks things.

Reply:
Does any of the Jasmine plugins modify `clearTimeout` and `setTimeout` functions? I had the exact same problem in [this PR](https://github.com/atom-community/atom-ide-outline/pull/118)

This code was working with Mocha, Sinon, Chai with not timing out.

Reply:
> Does any of the Jasmine plugins modify clearTimeout and setTimeout functions?

Not unless `jasmin.clock.install()` is called.

> Does Jasmine support async functions inside afterEach and beforeEach?

yes

> This code was not timing out with Mocha, Sinon, Chai

This was most likely because of a difference in sinon. `sinon.spy` calls the function by default and `spyOn` stubs the function by default.